connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaOffsetBackingStore.java

```diff
@@ -325,11 +325,12 @@ public Future<Void> set(final Map<ByteBuffer, ByteBuffer> values, final Callback
         return producerCallback;
     }
 
-    protected final Callback<ConsumerRecord<byte[], byte[]>> consumedCallback = new Callback<ConsumerRecord<byte[], byte[]>>() {
-        @Override
-        public void onCompletion(Throwable error, ConsumerRecord<byte[], byte[]> record) {
-            ByteBuffer key = record.key() != null ? ByteBuffer.wrap(record.key()) : null;
-            ByteBuffer value = record.value() != null ? ByteBuffer.wrap(record.value()) : null;
+    protected final Callback<ConsumerRecord<byte[], byte[]>> consumedCallback = (error, record) -> {
+        ByteBuffer key = record.key() != null ? ByteBuffer.wrap(record.key()) : null;
```

Comment:
I know this isn't a new bug introduced by this change, but we should probably add error-handling logic here:
```suggestion
        if (error != null) {
            log.error("Failed to read from offsets topic", error);
            return;
        }
        ByteBuffer key = record.key() != null ? ByteBuffer.wrap(record.key()) : null;
```

Reply:
While the suggestion makes sense and I've added it, it looks like the `error` will never be non-null as of now.

Reply:
Yes, I noticed that too; still, if the API we're using (even if it's internal-only) has a signature that can accept an error, we should handle those errors. An alternative is to refactor things to use a `java.util.function.Consumer` instead of a `Callback`, but it's not worth the impact that that would have on the `KafkaBasedLog` class.